MekHQ/src/mekhq/MekHQ.java

```diff
@@ -654,6 +667,158 @@ public IconPackage getIconPackage() {
+//                    Person person = getCampaign().getPerson(personId);
+//                    person.setHits(person.getHitsPrior());
+//                }
+                return;
+            }
+            PostScenarioDialogHandler.handle(
+                campaignGUI, getCampaign(), (AtBScenario) currentScenario, tracker, autoResolveConcludedEvent.controlledScenario());
+        } catch (Exception ex) {
+            logger.error("Error during auto resolve concluded", ex);
+        }
+    }
+
+
+    private static class SimulationScore {
+        private final AtomicInteger victories;
+        private final AtomicInteger losses;
+        private final AtomicInteger draws;
+        private final AtomicInteger gamesRun;
+
+        public SimulationScore() {
+            this.victories = new AtomicInteger(0);
+            this.losses = new AtomicInteger(0);
+            this.draws = new AtomicInteger(0);
+            this.gamesRun = new AtomicInteger(0);
+        }
+
+        public void addResult(AutoResolveConcludedEvent event) {
+            this.addResult(event.getGame());
+        }
+
+        public void addResult(IGame game) {
+            if (game.getVictoryTeam() == 1) {
+                victories.incrementAndGet();
+            } else if (game.getVictoryTeam() == 2) {
+                losses.incrementAndGet();
+            } else {
+                draws.incrementAndGet();
+            }
+            gamesRun.incrementAndGet();
+        }
+
+        public int getVictories() {
+            return victories.get();
+        }
+
+        public int getRuns() {
+            return gamesRun.get();
+        }
+
+        public int getLosses() {
+            return losses.get();
+        }
+
+        public int getDraws() {
+            return draws.get();
+        }
+    }
+
+    /**
+     * Calculates the victory chance for a given scenario and list of units by running multiple auto resolve scenarios in parallel.
+     *
+     * @param scenario the scenario to resolve
+     * @param units the list of units involved in the scenario
+     * @return the calculated victory chance as an integer percentage (0 to 100)
+     */
+    private SimulationScore calculateNumberOfVictories(int numberOfGames, AtBScenario scenario, List<Unit> units) {
+        var simulationScore = new SimulationScore();
+
+        if (numberOfGames <= 0) {
+            return simulationScore;
+        }
+
+        ExecutorService executor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+        List<Future<?>> futures = new ArrayList<>();
+        for (int i = 0; i < numberOfGames; i++) {
+            futures.add(executor.submit(() -> {
+                new Resolver(getCampaign(), units, scenario, new SimulationOptions(getCampaign().getGameOptions()), simulationScore::addResult)
+                    .resolveSimulation();
+            }));
+        }
+
+        // Wait for all tasks to complete
+        for (Future<?> future : futures) {
```

Comment:
Do we need to use a wildcard here, or can we use a specific type?

Reply:
It has to be a wildcard here unfortunatelly, but I think I can change one or another thing around and solve this.